restli-server/src/test/java/com/linkedin/restli/server/test/TestRestLiDefaultInResponse.java

```diff
@@ -0,0 +1,126 @@
+import com.linkedin.data.schema.resolver.FileDataSchemaResolver;
+import com.linkedin.data.schema.resolver.MultiFormatDataSchemaResolver;
+import com.linkedin.data.template.DataTemplateUtil;
+import com.linkedin.restli.internal.server.response.ResponseUtils;
+import java.io.File;
+import java.io.IOException;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.testng.reporters.Files;
+
+
+public class TestRestLiDefaultInResponse
+{
+  final static String FS = File.separator;
+  final static String testDir = System.getProperty("testDir", new File("src/test").getAbsolutePath());
+  final static String pegasusDir = testDir + FS + "pegasus" + FS + "com" + FS + "linkedin" + FS + "restli" + FS + "server" + FS + "defaults";
+  final static String resolverDir = testDir + FS + "pegasus";
+
+  @BeforeTest
+  public void beforeTest()
+  {
+    System.setProperty(AbstractGenerator.GENERATOR_RESOLVER_PATH, resolverDir);
+  }
+
+  @DataProvider(name = "default_serialization")
+  public Object[][] schemaFilesForDefaultSerializationTest()
+  {
+    DataList recordBField1 = new DataList(1);
+    recordBField1.add(-1);
+
+    DataMap recordBField2 = new DataMap();
+    DataList recordBField2DefaultValues = new DataList(2);
+    recordBField2DefaultValues.add("defaultValue1");
+    recordBField2DefaultValues.add("defaultValue2");
+    recordBField2.put("defaultKey", recordBField2DefaultValues);
+    DataMap personalRecordD = new DataMap();
+    personalRecordD.put("field3", 0L);
+    personalRecordD.put("field4", "default");
+
+    return new Object[][]{
+        {
+            "RecordA.pdl",
+            new DataMap(new ImmutableMap.Builder<String, Object>()
+                .put("field1", 1)
+                .put("field2", "2")
+                .build()),
+            new DataMap(new ImmutableMap.Builder<String, Object>()
+                .put("field1", 1)
+                .put("field2", "2")
+                .put("field3", 0L)
+                .put("field4", "default")
+                .build())
+        },
+        {
+            "RecordB.pdl",
+            new DataMap(new ImmutableMap.Builder<String, Object>()
+                .build()),
+            new DataMap(new ImmutableMap.Builder<String, Object>()
+                .put("field1", recordBField1)
+                .put("field2", recordBField2)
+                .build())
+        },
+        {
+            "RecordC.pdl",
+            new DataMap(),
+            new DataMap(new ImmutableMap.Builder<String, Object>()
+                .put("name", "default+")
+                .put("personalRecordD", personalRecordD)
+                .build()),
+        }
+    };
+  }
+
+  @Test(dataProvider = "default_serialization")
+  public void testSerializingDefaultValue(String filename, DataMap data, DataMap expected)
```

Comment:
Consider renaming this to `TestGetAbsentFieldsDefaultValues` to be aligned with the method it's testing. If you rename that method, rename this test accordingly.